src/main/java/org/folio/rest/impl/PostOrderLineHelper.java

```diff
@@ -95,149 +80,6 @@ private void addSuffixToPoLineNumber(JsonObject line, JsonObject sequenceNumber)
-          compPOL.getEresource().setId(id);
-          compPOL.getEresource().setPoLineId(compPOL.getId());
-        }
-      })
-      .exceptionally(t -> {
-        logger.error("failed to create EResource", t);
-        throw new CompletionException(t.getCause());
-      });
-  }
-
-  private CompletableFuture<Void> createLocations(CompositePoLine compPOL, JsonObject line) {
-    List<Location> locations = compPOL.getLocations();
-    if (CollectionUtils.isNotEmpty(locations)) {
-      List<CompletableFuture<String>> futures = new ArrayList<>();
-      Iterator<Location> iterator = locations.iterator();
-      while (iterator.hasNext()) {
-        Location location = iterator.next();
-        futures.add(createSubObjIfPresent(line, location, LOCATIONS, resourcesPath(LOCATIONS))
-          .thenApply(id -> {
-            if (StringUtils.isEmpty(id)) {
-              iterator.remove();
-            } else {
-              location.setId(id);
-              location.setPoLineId(compPOL.getId());
-            }
-            return id;
-        }));
-      }
-
-    return collectResultsOnSuccess(futures)
-      .thenAccept(ids -> {
-        if (CollectionUtils.isEmpty(ids)) {
-          line.remove(LOCATIONS);
-          compPOL.setLocations(Collections.emptyList());
-        } else {
-          line.put(LOCATIONS, ids);
-        }
-      })
-      .exceptionally(t -> {
-        logger.error("Failed to create Locations", t);
-        throw new CompletionException(t.getCause());
-      });
-    } else {
-      line.remove(LOCATIONS);
-    }
-
-    return completedFuture(null);
-  }
-
-  private CompletableFuture<Void> createPhysical(CompositePoLine compPOL, JsonObject line) {
-    return createSubObjIfPresent(line, compPOL.getPhysical(), PHYSICAL, resourcesPath(PHYSICAL))
-      .thenAccept(id -> {
-        if (id == null) {
-          line.remove(PHYSICAL);
-          compPOL.setPhysical(null);
-        } else {
-          compPOL.getPhysical().setId(id);
-          compPOL.getPhysical().setPoLineId(compPOL.getId());
-        }
-      })
-      .exceptionally(t -> {
-        logger.error("failed to create Physical", t);
-        throw new CompletionException(t.getCause());
-      });
-  }
-
-  private CompletableFuture<Void> createVendorDetail(CompositePoLine compPOL, JsonObject line) {
-    return createSubObjIfPresent(line, compPOL.getVendorDetail(), VENDOR_DETAIL, resourcesPath(VENDOR_DETAIL))
-      .thenAccept(id -> {
-        if (id == null) {
-          line.remove(VENDOR_DETAIL);
-          compPOL.setVendorDetail(null);
-        } else {
-          compPOL.getVendorDetail().setId(id);
-          compPOL.getVendorDetail().setPoLineId(compPOL.getId());
-        }
-      })
-      .exceptionally(t -> {
-        logger.error("failed to create VendorDetail", t);
-        throw new CompletionException(t.getCause());
-      });
-  }
-
   private CompletableFuture<Void> createReportingCodes(CompositePoLine compPOL, JsonObject line) {
```

Comment:
Also WRT, https://github.com/folio-org/acq-models/pull/89, please check if we are supposed to create reporting codes at all while placing an order

Reply:
After talking to @craigmcnally, it seems like we should not be creating and deleting reporting_code. So, I will remove this code. Thanks for pointing it out.

Reply:
For now, reporting_code will remain as is. At some point, this will be replaced by tags.